include/boost/math/tools/polynomial.hpp

```diff
@@ -134,7 +236,11 @@ class polynomial
 
    // access:
    size_type size()const { return m_data.size(); }
-   size_type degree()const { return m_data.size() - 1; }
+   size_type degree()const
+   { 
+       BOOST_ASSERT(size() != 0);
```

Comment:
A last thought before I go to bed: if we provide the most efficient implementation with the least safeguards, it can easily be wrapped in functions that offer more safeguards. Doing the opposite (removing safeguards for added efficiency) is non-trivial.